src/app/error-handling/middlewares/error-handler.middleware.js

```diff
@@ -0,0 +1,26 @@
+const { mapErrors } = require('../../utils');
+const HttpStatus = require('http-status-codes');
+
+const middleware = ({ logger }) => (err, req, res, next) => {
+  let status = err.status;
+  let message = err.message;
+  let messages = err.messages || '';
+  if (!err.status) {
+    switch (err.name) {
```

Comment:
It would be better to check error's instance or type. Here you can use strategy pattern or similar.